manager_synchronized.go

```diff
@@ -53,6 +51,8 @@ func (m *synchronizedProcessManager) CreateProcess(ctx context.Context, opts *op
 		return nil, errors.WithStack(err)
 	}
 
+	opts.Synchronized = true
```

Comment:
I think modifying this here, after the process is created is probably wrong.

Reply:
I think we can just delete this line, and then merge it later, and revisit later...

Reply:
Putting it before is going to break any possibility of having multithreading in remote services. If you do this, all remote service which must call Get() will use the exact same synchronized process and will be forced to do entirely single-threaded operations on processes.

Reply:
I think that's right though? On the other hand you end up with a situation where you say "give me a locked thing" you're asking for singally-threaded-access to a process?

Reply:
It's not though, because the assumption has always been that you can perform multiple operations on a given process. If somebody was waiting on the process in one goroutine, it doesn't make sense that the wait would block all other operations from being processed (e.g. you can't get process info, you can't ask it if it's done, all you can do is wait for it to finish). All the remote services use the synchronized manager so this would be bad as far as usability of remote services.

Reply:
It also has implications for potentially blocking all manager functionality which calls any method on a process. If the processes are all stuck waiting for other operations to finish which could take a long time (e.g. a wait or a signal), all your process-related requests might be locked out (e.g. a manager won't be able to close because it's blocked trying to signal processes to exit).